### PR TITLE
Build envoy as dependency

### PR DESCRIPTION
**Pull request by lizan**

### CODE REVIEW

File: repositories.bzl [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+            actual = "@protobuf_git//:protoc_lib",
+        )
+
+        native.bind(
+            name = "protobuf_clib",
+            actual = "@protobuf_git//:protobuf_lite",
+        )
+
+
+def googletest_repositories(bind=True):
```
Comment (chowchow316) [inline repositories.bzl]: Why we move this repository here?

File: repositories.bzl [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            actual = "@boringssl//:crypto",
+        )
+
+        native.bind(
+            name = "libssl",
+            actual = "@boringssl//:ssl",
+        )
+
+
+def protobuf_repositories(bind=True):
```
Comment (chowchow316) [inline repositories.bzl]: We have one protobuf_repositories in contrib/endpoints/repositories.bzl, can we merge them?

Comment (lizan) [inline repositories.bzl]: forgot to save that file...

File: repositories.bzl (hunk 1 above)
Comment (lizan) [inline repositories.bzl]: Envoy uses googletest for test too.

Comment (chowchow316) [inline repositories.bzl]: So this file contains the common dependencies of ESP and envoy?

Comment (lizan) [inline repositories.bzl]: right

**Pull request merged**